git-ftp

```diff
@@ -818,8 +818,13 @@ submodule_catchup() {
 }
 
 set_submodule_args() {
-	# Duplicate the current required parameters
-	args=(--user "$REMOTE_USER" --passwd "$REMOTE_PASSWD")
+	args=(--user "$REMOTE_USER")
+	if [ "$REMOTE_PROTOCOL" = "sftp" ] && [ -n "$CURL_PRIVATE_KEY" ] && [ -n "$CURL_PUBLIC_KEY" ]; then
```

Comment:
I'm not sure but I thought that the keys can be auto-detected on some platforms. At least the public key can be detected with modern curl. This line would fail if it's not given. Can we instead just check which options are set and pass them on? Maybe we now think it doesn't make sense to set a key and a password but if they are both set, we should pass them on. Logic may change. Maybe one day in the future somebody wants to use the password for protected keys and then needs both.

Reply:
**Testing**
Yes you are completely right. I’ll add a test for the --insecure flag. 

At the moment sftp is not tested at all as well as the handling of keys. Testing sftp is quite complex since curl does not support this by default, at least on macOS. I’m already testing some stuff to install a custom build curl and allow ssh access during the install script. But test for sftp is another topic I would propose to do this in another branch.

**Password and Key**
Yes this a interesting thought. We have some more issues according to password handling (#309 #310 #349 and maybe more). I think you should find a solution that covers all o them. But you are right. At the moment we should go for a similar behaviour as in the main task.